### ISSUE TITLE
[508-defect-1 ❗] Alert and status messages aren't announced without receiving focus.

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#42249]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _First name only. Delete if not applicable._

## User Story or Problem Statement

As a screen reader user navigating through a form, I will miss key alert and status messages that dynamically appear visually without being announced or accessible by the `tab` key or up and down form control swipe actions on iOS VoiceOver

## Details

Upon completing the date of burial, a yellow alert box dynamically appears without being announced to screen reader users. This results in several issues including:
- Screen reader users will not know this alert exists as it is sandwiched between form controls. Using the `tab` key will only surface the link.
- Magnification users may be disorientated and unaware that the link opens a new tab instead of changing the current window as no warning is provided ahead of time ([WCAG G201: G201: Giving users advanced warning when opening a new window](https://www.w3.org/TR/WCAG20-TECHS/G201.html))
- Users with colorblindness may be unaware of the urgency of the message as the yellow background color alone is used to indicate urgency

<img width="1211" alt="Screen Shot 2022-06-02 at 3 22 05 PM" src="https://user-images.githubusercontent.com/14154792/171724350-d5e15950-d4f2-435e-9fbe-ff11aaac7756.png">

https://user-images.githubusercontent.com/14154792/171724373-d9fe66bd-54c7-4f95-92a1-52d1e7bbfcfb.mov

## Acceptance Criteria

- [ ] The dynamic alert is announced using `aria-live="polite"`
- [ ] An icon with alt text or a text label is provided to represent the urgency the background color yellow communicates (e.g. "**Notice**: If filing for a non-service-connected allowance, the Veteran’s burial date must be no more than 2 years from the current date. Find out if you still qualify..."
- [ ] The string "If filing for a non-service-connected allowance, the Veteran’s burial date must be no more than 2 years from the current date. Find out if you still qualify" describes the link "Learn about eligibility"
- [ ] The link text should include "(opens in new tab)" to give magnification users a heads up that it will be opening in a new tab

## Steps to Recreate

1. Enter `[<IP ADDRESS>](https://staging.va.gov/burials-and-memorials/application/530/veteran-information/burial)` in browser
2. Enter January 1, 2000 as the date of birth
3. Enter January 1, 2001 as the date of death
4. Turn on your screen reader device
5. Enter January 1, 2002 as the date of burial
6. Confirm the yellow message is not announced via screen reader
7. Click on the link within the yellow box, confirm that it opens in a new tab without warning the user

## Proposed Solution
Follow acceptance criteria

<img width="1227" alt="Screen Shot 2022-06-02 at 3 42 04 PM" src="https://user-images.githubusercontent.com/14154792/171725080-7f4a4bd5-ca49-467a-8b41-ac1ee1b15b01.png">

## WCAG or Vendor Guidance (optional)

* [G201: Giving users advanced warning when opening a new window](https://www.w3.org/TR/WCAG20-TECHS/G201.html)
* [11.2 Platform standards](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/va-gov-experience-standards)

**Comment by joshkimux** [comment]
@jerekshoe , va-alert would do the trick provided it's using a polite `aria-live` region!

As for the link, it depends, and is more of a UX question 😆 

If they're logged in, and leaving the page may make the user feel like they've lost their session or actually lose their session, than we should use (open in new tab). Otherwise, if they won't lose their session and can easily (and confidently) return to it using the back button, we can simply not have it open a new tab.

**Comment by joshkimux** [comment]
<img width="877" alt="Screen Shot 2022-06-30 at 8 23 56 AM" src="https://user-images.githubusercontent.com/14154792/176676577-2618349e-c3bb-43bc-ba6d-69a3309d1541.png">

Tested and confirmed this is working as expected! 💯
